### ISSUE TITLE
[BUG - All] Can't Use Lua and C++ At Same Time

### ISSUE THREAD

**Issue by Mythical-Github** [UE4SS-RE/RE-UE4SS#161]
**Branch or Release**
b69056953a1abb279560724c1d1dd5801fd0ffb9

**Game and Engine Version**
EscapeTheBackrooms UE 4.27

**Describe the bug**
When you have a mod enabled that uses both lua files and dlls, it will only load the lua files, and not the dll

**To Reproduce**
Make an example lua mod, and dll mod
and try to load both at the same time with the same name
I used ExampleMod for mine

**Expected behavior**
Both Load

**OS**
 Windows 10

**Comment by CpazR** [comment]
Noticed this too. Would be nice to create custom, mod-specific LUA bindings through DLLs, but this seems to be preventing that, unfortunately.

**Comment by UE4SS** [comment]
Fixed by 435d6891208d4c5fa165d3072cffcb55747c1b41

**Comment by UE4SS** [comment]
> Tried this using Kingdom Hearts 3 and I noticed this too. Would be nice to create custom, mod-specific LUA bindings through DLLs, but this seems to be preventing that, unfortunately.

This is now possible as of 4e3cb15bbc71432e96485e41fed00e107a9f8c85
Look at `CppUserModBase.hpp`, and here's an example that registers a global function for this mod:
```c++
auto on_lua_start(LuaMadeSimple::Lua& lua, LuaMadeSimple::Lua& main_lua, LuaMadeSimple::Lua& async_lua, std::vector<LuaMadeSimple::Lua*>& hook_luas) -> void override
{
    Output::send(STR("Lua Mod started\n"));
    lua.register_function("MyTestFunction", [](const LuaMadeSimple::Lua& lua) -> int {
        Output::send(STR("MyTestFunction called from a Lua mod\n"));
        return 0;
    });
    Output::send(STR("C++ Mod registered MyTestFunction\n"));
}
```
If you'd rather work directly with `lua_State*`, then you can call `lua.get_lua_state()`.
Note that this callback is only fired when a mod by the same name as the C++ mod is started, and the Lua instances are specific to that particular Lua mod as well.

**Comment by CpazR** [comment]
> This is now possible as of [4e3cb15](https://github.com/UE4SS-RE/RE-UE4SS/commit/4e3cb15bbc71432e96485e41fed00e107a9f8c85) Look at `CppUserModBase.hpp`, and here's an example that registers a global function for this mod:
> 
> ```c++
> auto on_lua_start(LuaMadeSimple::Lua& lua, LuaMadeSimple::Lua& main_lua, LuaMadeSimple::Lua& async_lua, std::vector<LuaMadeSimple::Lua*>& hook_luas) -> void override
> {
>     Output::send(STR("Lua Mod started\n"));
>     lua.register_function("MyTestFunction", [](const LuaMadeSimple::Lua& lua) -> int {
>         Output::send(STR("MyTestFunction called from a Lua mod\n"));
>         return 0;
>     });
>     Output::send(STR("C++ Mod registered MyTestFunction\n"));
> }
> ```
> 
> If you'd rather work directly with `lua_State*`, then you can call `lua.get_lua_state()`. Note that this callback is only fired when a mod by the same name as the C++ mod is started, and the Lua instances passed to it are specific to that particular Lua mod as well.

This works well! However, It looks like restarting mods results in the `on_lua_start` method to be called after the lua script has ran? It works on initial startup, but it can't find the custom method on a hot-reload of the lua.

Not sure if I've missed something or not.

**Comment by UE4SS** [comment]
> > This is now possible as of [4e3cb15](https://github.com/UE4SS-RE/RE-UE4SS/commit/4e3cb15bbc71432e96485e41fed00e107a9f8c85) Look at `CppUserModBase.hpp`, and here's an example that registers a global function for this mod:
> > ```c++
> > auto on_lua_start(LuaMadeSimple::Lua& lua, LuaMadeSimple::Lua& main_lua, LuaMadeSimple::Lua& async_lua, std::vector<LuaMadeSimple::Lua*>& hook_luas) -> void override
> > {
> >     Output::send(STR("Lua Mod started\n"));
> >     lua.register_function("MyTestFunction", [](const LuaMadeSimple::Lua& lua) -> int {
> >         Output::send(STR("MyTestFunction called from a Lua mod\n"));
> >         return 0;
> >     });
> >     Output::send(STR("C++ Mod registered MyTestFunction\n"));
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > If you'd rather work directly with `lua_State*`, then you can call `lua.get_lua_state()`. Note that this callback is only fired when a mod by the same name as the C++ mod is started, and the Lua instances passed to it are specific to that particular Lua mod as well.
> 
> This works well! However, It looks like restarting mods results in the `on_lua_start` method to be called after the lua script has ran? It works on initial startup, but it can't find the custom method on a hot-reload of the lua.
> 
> Not sure if I've missed something or not.

Whoops. Turns out we were loading Lua mods before C++ mods specifically when hot-reloading.
This has been fixed by cf1fc1ae178ab0bed3065c7321d0128a681169ac
